mlflow/tracking/client.py

```diff
@@ -598,15 +601,90 @@ def search_prompts(
+                    INVALID_PARAMETER_VALUE,
+                )
+            
+            catalog_name, schema_name, remaining_filter = self._parse_catalog_schema_filter(filter_string)
+            
+            if not catalog_name or not schema_name:
+                raise MlflowException(
+                    "For Unity Catalog prompt registries, you must specify both catalog and schema "
+                    "in the filter string: catalog = 'catalog_name' AND schema = 'schema_name'",
+                    INVALID_PARAMETER_VALUE,
+                )
+            
+            if remaining_filter:
+                raise MlflowException(
+                    f"Unity Catalog prompt search currently only supports catalog and schema filters. "
+                    f"Unsupported filter: {remaining_filter}",
+                    INVALID_PARAMETER_VALUE,
+                )
+            
+            # Call Unity Catalog store directly with structured parameters
+            return registry_client.store.search_prompts(
+                catalog_name=catalog_name,
+                schema_name=schema_name,
+                filter_string=remaining_filter,
+                max_results=max_results,
+                page_token=page_token,
+            )
+        else:
+            # For traditional registries, use existing prompt tag logic
+            fls = f"tag.`{IS_PROMPT_TAG_KEY}` = 'true'"
+            if filter_string:
+                fls = f"{fls} AND {filter_string}"
+
+            return registry_client.search_registered_models(
+                filter_string=fls,
+                max_results=max_results,
+                page_token=page_token,
+            )
 
-        return self._get_registry_client().search_registered_models(
-            filter_string=fls,
-            max_results=max_results,
-            page_token=page_token,
-        )
+    def _parse_catalog_schema_filter(self, filter_string: str) -> tuple[Optional[str], Optional[str], Optional[str]]:
+        """
+        Parse catalog and schema from filter string for Unity Catalog.
+        
+        Args:
+            filter_string: Filter string potentially containing catalog and schema
+            
+        Returns:
+            Tuple of (catalog_name, schema_name, remaining_filter)
+        """
+        if not filter_string:
+            return None, None, None
+            
+        # Patterns to match catalog and schema (case insensitive, single or double quotes)
+        catalog_pattern = r'catalog\s*=\s*[\'"]([^\'"]+)[\'"]'
+        schema_pattern = r'schema\s*=\s*[\'"]([^\'"]+)[\'"]'
+        
+        catalog_match = re.search(catalog_pattern, filter_string, re.IGNORECASE)
+        schema_match = re.search(schema_pattern, filter_string, re.IGNORECASE)
+        
+        catalog_name = catalog_match.group(1) if catalog_match else None
+        schema_name = schema_match.group(1) if schema_match else None
```

Comment:
Can we use `sqlparse` for this? See examples in https://github.com/mlflow/mlflow/blob/master/mlflow/utils/search_utils.py?